### PR TITLE
fix (sysadvisor): pap - fixes of defects identified in integration validation

### PR DESCRIPTION
**Pull request by h-w-chen**
#### What type of PR is this?
Bug fixes

#### What this PR does / why we need it:
this PR has 3 fixes:
1.  explicit internal-op over autonomy in last 2 minutes (behavior change);
2. power cap sock file folders creation in case it is not there;
3. split power cap server and plugin to avoid recursive reference chain via plugin wrapper

### CODE REVIEW

File: pkg/agent/sysadvisor/plugin/poweraware/advisor/action/strategy/strategy.go [hunk 1]
```diff
@@ -57,7 +57,10 @@ func (p ruleBasedPowerStrategy) RecommendAction(actualWatt int,
 
 	if ttl <= time.Minute*2 {
 		// whatever valid alert, power capping should do in short of 2 minutes
-		return action.PowerAction{Op: spec.InternalOpFreqCap, Arg: desiredWatt}
+		// this the default behavior unless instructed explicitly by internal op
```
Comment (gary-lgy) [inline pkg/agent/sysadvisor/plugin/poweraware/advisor/action/strategy/strategy.go:60]: Should this be part of the `autoAction` function?

File: pkg/agent/sysadvisor/plugin/poweraware/evictor/server/evictor_service.go [hunk 2]
```diff
@@ -110,16 +110,9 @@ func (p *powerPressureEvictServer) Start() error {
...
-
-	if !p.started {
-		general.InfofV(6, "pap: power pressure eviction server already stopped")
-		return nil
-	}
-
-	p.started = false
-	return p.service.Stop()
+	// since there is no resource other than the underlying PluginRegistrationWrapper-guarded grpc service,
+	// which will be cleanup by PluginRegistrationWrapper itself, it has nothing to clean up here
```
Comment (gary-lgy) [inline pkg/agent/sysadvisor/plugin/poweraware/evictor/server/evictor_service.go:114]: `PluginRegistrationWrapper` has an internal stop channel that controls the stopping of the server. How do we close the stop channel without calling `Stop` on the wrapper?

File: pkg/agent/sysadvisor/plugin/poweraware/advisor/action/strategy/strategy.go (hunk 1 above)
Comment (h-w-chen) [reply pkg/agent/sysadvisor/plugin/poweraware/advisor/action/strategy/strategy.go:60]: yes it is actually. will update code NOT to return directly here, and let autoAction to fill up the needed detail.

File: pkg/agent/sysadvisor/plugin/poweraware/evictor/server/evictor_service.go (hunk 2 above)
Comment (h-w-chen) [reply pkg/agent/sysadvisor/plugin/poweraware/evictor/server/evictor_service.go:114]: good point! the current implementation has a recursive reference, which leads to recursive calls of Stop. Will reconstruct code to break the loop.

File: pkg/agent/sysadvisor/plugin/poweraware/evictor/server/evictor_service.go [hunk 3]
```diff
@@ -43,24 +42,47 @@ const (
 var errPowerPressureEvictionPluginUnavailable = errors.New("power pressure eviction plugin is unavailable")
 
 type powerPressureEvictServer struct {
+	plugin  *powerPressureEvictPlugin
```
Comment (gary-lgy) [inline pkg/agent/sysadvisor/plugin/poweraware/evictor/server/evictor_service.go:45]: Where is `plugin` initialized?

Comment (h-w-chen) [reply pkg/agent/sysadvisor/plugin/poweraware/evictor/server/evictor_service.go:45]: thanks for catching this - will add it as server object field

**Review by gary-lgy — APPROVED** [review]

**Pull request merged**